spreadsheet/sheet.go

```diff
@@ -871,3 +872,256 @@ func (s *Sheet) Sort(column string, firstRow uint32, order SortOrder) {
 		}
 	}
 }
+
+// RemoveColumn removes column from the sheet and moves all columns to the right of the removed column one step left.
+func (s *Sheet) RemoveColumn(column string) error {
+	cellsInFormulaArrays, err := s.getAllCellsInFormulaArraysForColumn()
+	if err != nil {
+		return err
+	}
+	columnIdx := reference.ColumnToIndex(column)
+	for _, row := range s.Rows() {
+		ref := fmt.Sprintf("%s%d", column, *row.X().RAttr)
+		if _, ok := cellsInFormulaArrays[ref]; ok {
+			return nil
+		}
+	}
+	for _, row := range s.Rows() {
+		cells := row.x.C
+		for ic, cell := range cells {
+			ref, err := reference.ParseCellReference(*cell.RAttr)
+			if err != nil {
+				return err
+			}
+			if ref.ColumnIdx == columnIdx {
+				row.x.C = append(cells[:ic], s.slideCellsLeft(cells[ic+1:])...)
+				break
+			} else if ref.ColumnIdx > columnIdx {
+				row.x.C = append(cells[:ic], s.slideCellsLeft(cells[ic:])...)
+				break
+			}
+		}
+	}
+
+	err = s.moveLeftFormulasArgs(columnIdx)
```

Comment:
would it make sense to have an argument `updateReferences bool` and only execute the following ones
if updateReferences is true?

Reply:
I think there is no significant need because it seems to be an expensive operation only if a workbook has a lot of sheets/cells, but it is hardly possible for such a document to not to have formulas. Also extra arguments make functions more complicated. But I can do if it is needed.

Reply:
ok